builtin/v12/migration/miner.go

```diff
@@ -28,10 +33,12 @@ type minerMigrator struct {
 	emptyDeadlinesV11 cid.Cid
 	emptyDeadlineV12  cid.Cid
 	emptyDeadlinesV12 cid.Cid
+	sectorDeals       *builtin.ActorTree
 	OutCodeCID        cid.Cid
+	hamtLock          *sync.Mutex
```

Comment:
nit: since there are a lot of hamts in the miner actor give this a slightly more descriptive name i.e. sectorDealsLock.  Also consider renaming sectorDeals to something that better communicates its global nature (i.e. marketSectorDeals, globalSectorDeals, allSectorDeals..)

Reply:
Also Im quite confused how this migration is working at all since hamtLock never appears to be initialized.  You should be getting a panic when you call this: https://go.dev/play/p/_7PCDPsWrac